core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalReader.java

```diff
@@ -247,4 +253,23 @@ private void updateInputStream() throws IOException {
       mInputStream = new JournalInputStream(mFilesToProcess.poll());
     }
   }
+
+  @Override
+  public State advance() throws IOException {
+    mCheckpointStream = null;
+    mNextEntry = null;
+    while (true) {
+      updateInputStream();
+      if (mCheckpointStream != null) {
+        return State.CHECKPOINT;
+      }
+      advanceEntry();
+      if (mNextEntry != null) {
+        return State.LOG;
+      }
+      if (mInputStream == null || mInputStream.isDone() || mInputStream.mFile.isIncompleteLog()) {
```

Comment:
if `mNextEntry` is `null`, will this if statement always evaluate to `true`? If not, then it would just loop again, and what would happen?

Reply:
removed the outer while loop